colorizer_data/types.py

```diff
@@ -97,15 +100,64 @@ class FrameDimensions(TypedDict):
     """Height of a frame in physical units (not pixels)."""
 
 
-class DatasetMetadata(TypedDict):
+class BaseMetadataJson(TypedDict):
+    """JSON dictionary format for `BaseMetadata`."""
+
+    name: str
+    description: str
+    author: str
+    dateCreated: str
+    lastModified: str
+    revision: str
+    dataVersion: str
+
+
+@dataclass
+class BaseMetadata:
+    """Shared metadata between datasets and collection files."""
+
+    name: str = None
+    description: str = None
+    author: str = None
+    date_created: str = None
+    """Formatted datetime string. See `DATETIME_FORMAT`. """
+    last_modified: str = None
+    """Formatted datetime string. See `DATETIME_FORMAT`. """
```

Comment:
is the datetime format consistent with what JS code can parse?

Reply:
This is a good point, I'll update this to use the official JavaScript specification for simplicity

Reply:
Learning right now that there's a whole can of worms behind timestamp formatting, but the new format should be ISO standard.